File: services/credit-service/src/main/java/com/gti_e_credit/credit/service/credit/Credit.java
package com.gti_e_credit.credit.service.credit;
import com.gti_e_credit.credit.service.document.Documents;
import jakarta.annotation.Nullable;
import jakarta.persistence.*;
import lombok.*;

import java.math.BigInteger;
import java.util.List;


@Entity
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Builder

public class Credit {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;
    private String typeCredit;


    @OneToMany(mappedBy = "credit")
    private List<Documents> documents;
    private BigInteger MaxMontant;

    private float interestRate;
    private Integer maxNbrDecheance;

}

File: creditUi/src/app/services/models/credit.ts
export interface documents{
    id:number;
    name:string;
    description:string;
    isRequired:boolean;
}




export interface Credit{
    id: number;
    typeCredit:string;
    MaxMontant:number;
    maxNbrDecheance:number;
    interestRate:number; 
    description:string;
    documents:documents[]; 

}